Connectors/Rhino/Speckle.Connectors.GrasshopperShared/HostApp/DocChangeMonitor.cs

```diff
@@ -0,0 +1,95 @@
+using Rhino;
+using Speckle.Connectors.GrasshopperShared.HostApp.Extras;
+using Speckle.Sdk;
+
+namespace Speckle.Connectors.GrasshopperShared.HostApp;
+
+/// <summary>
+/// Monitors Rhino doc events (creating, opening, switching) and notifies subscribed components when documents change.
+/// These aren't caught by Grasshopper's DocumentContextChanged.
+/// </summary>
+/// <remarks>
+/// Not tied to any specific component - doc events are global. When any doc changes, all interested components should know.
+/// </remarks>
+public static class DocChangeMonitor
```

Comment:
This is an EventAggregator.  Using statics with WeakReferences is fine but I'd rather DI was involved.

I'm more pointing this out because we removed an earlier implementation because of threading issues.

Reply:
Linked to above, i suppose